pipeline_steps/common.groovy

```diff
@@ -1672,7 +1669,7 @@ void globalWraps(Closure body){
   // global timeout is long, so individual jobs can set shorter timeouts and
   // still have to cleanup, archive atefacts etc.
   timestamps{
-    timeout(time: 10, unit: 'HOURS'){
+    timeout(time: env.BUILD_TIMEOUT_HRS, unit: 'HOURS'){
```

Comment:
You could get this to parse gate by setting a default for when the env var isn't available. 
```suggestion
    timeout(time: env.BUILD_TIMEOUT_HRS ?: 3, unit: 'HOURS'){
```